### DISCUSSION TITLE
Nextjs 13 How to render completely on the client side?

### DISCUSSION THREAD

**Discussion by dejurin** [vercel/next.js#42240] [Help]
How to render on the client side in the nextjs 13 with **appDir=true conf**?

I need to render a component that will run completely on the client side, without server rendering?
Before, I used **next/dynamic**, but with **appDir=true** config it doesn't work.

**Comment by premier213** [comment]
use 
'use client' at the top of file

such as
![image](https://user-images.githubusercontent.com/4321174/199047244-9190b861-ff58-45c6-8e7d-bb2bc6fa2669.png)

**Reply by dejurin** [reply]
I tried, nope.

**Comment by icyJoseph** [comment]
I think you are better off with a `show` flag combo of `useState` + `useEffect`.

Even if you use a client directive the component Next.js pre-renders them on the server:

> Client Components are rendered on the client. With Next.js, Client Components can also be [pre-rendered](https://beta.nextjs.org/docs/rendering/fundamentals#client-and-server-rendering-at-the-component-level) on the server and [hydrated](https://beta.nextjs.org/docs/rendering/fundamentals#client-and-server-rendering-at-the-component-level) on the client.

You can still have the dynamic bit I think, to keep it as a separate chunk, but to exclusively run on the client, the `show` flag would have to be used.

Since it'd use `useState` and `useEffect` you need to put the `use client` directive anyway, but then hide it behind `if(!show) return null`. However you'll observe that code above the `if`-statement is still run on the server.

I am not sure if there's a more elegant, Server/Client components mental model inspired, solution, but this is a workaround for the time being I guess.

Also, perhaps, we must ask ourselves if this new mental model solves bundling issues we had solved through code-splitting. For JSX that strictly depends on DOM APIs or unsafe DOM access, the show flag ought to be enough.
